nessie/run.go

```diff
@@ -0,0 +1,124 @@
+package nessie
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/thanhpk/randstr"
+	"github.com/treeverse/lakefs/api"
+	"github.com/treeverse/lakefs/api/gen/client"
+	"github.com/treeverse/lakefs/api/gen/client/setup"
+	"github.com/treeverse/lakefs/api/gen/models"
+	"github.com/treeverse/lakefs/logging"
+)
+
+// Config contains configuration needed for running the tests
+type Config struct {
+	// BaseURL is the base address of the lakeFS endpoint
+	BaseURL string
+
+	// BucketPath is the full path to the s3 bucket
+	BucketPath string
+}
+
+// Run runs system tests and reports on failures
+func Run(ctx context.Context, config Config) error {
+	// initialize the env/repo
+	logger := logging.FromContext(ctx)
+	bucketPath := fmt.Sprintf("%s/%s", config.BucketPath, time.Now().UTC().Format("2006/01/02/150405"))
+	logger.WithField("bucketPath", bucketPath).Info("Starting nessie run")
+
+	cl := client.NewHTTPClientWithConfig(strfmt.Default, &client.TransportConfig{
+		Host:     config.BaseURL,
+		BasePath: client.DefaultBasePath,
+		Schemes:  []string{"http"},
```

Comment:
I'd make this configurable to support running against a "real" system that might not allow http?